### PR TITLE
8314832: Few runtime/os tests ignore vm flags

### PR DESCRIPTION
**Pull request by mseledts**
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8314832](https://bugs.openjdk.org/browse/JDK-8314832): Few runtime/os tests ignore vm flags (**Sub-task** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15647/head:pull/15647` \
`$ git checkout pull/15647`

Update a local copy of the PR: \
`$ git checkout pull/15647` \
`$ git pull https://git.openjdk.org/jdk.git pull/15647/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15647`

View PR using the GUI difftool: \
`$ git pr show -t 15647`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15647.diff">https://git.openjdk.org/jdk/pull/15647.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15647#issuecomment-1712304273)

### CODE REVIEW

File: test/hotspot/jtreg/runtime/os/THPsInThreadStackPreventionTest.java
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/os/THPsInThreadStackPreventionTest.java:1]: I think this test should also be marked vm.flagless. Running with additional flags, especially something like -Xcomp, could have an effect on the RSS usage that the test is trying to measure. Other flags really have no bearing on what is being tested here.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]

File: test/hotspot/jtreg/runtime/os/THPsInThreadStackPreventionTest.java
Comment (mseledts) [reply test/hotspot/jtreg/runtime/os/THPsInThreadStackPreventionTest.java:1]: David, thank you for advice. I will update the test to make it "flagless".

File: test/hotspot/jtreg/runtime/os/THPsInThreadStackPreventionTest.java [hunk 1]
```diff
@@ -40,6 +41,7 @@
  * @bug 8303215 8312182
  * @summary On THP=always systems, we prevent THPs from forming within thread stacks (negative test)
  * @library /test/lib
+ * @requires vm.flagless
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/os/THPsInThreadStackPreventionTest.java:44]: Not sure this is needed here as this is a manual test

File: test/hotspot/jtreg/runtime/os/TestTrimNative.java [hunk 2]
```diff
@@ -36,6 +37,7 @@
 
 /*
  * @test id=trimNativeStrict
+ * @requires vm.flagless
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/os/TestTrimNative.java:40]: Ditto perhaps not needed for a manual test

File: test/hotspot/jtreg/runtime/os/TestTrimNative.java [hunk 3]
```diff
@@ -69,6 +73,7 @@
 /*
  * @test id=trimNativeLowIntervalStrict
  * @summary Very low (sub-second) interval, nothing should explode (stricter test, manual mode)
+ * @requires vm.flagless
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/os/TestTrimNative.java:76]: Ditto manual test

**Review by dholmes-ora — APPROVED** [review]
I don't think we need to add flagless to the manual tests, but I'm not sure. I will leave it to you.

Thanks

**Review by lmesnik — APPROVED** [review]

**Comment by mseledts** [comment]
Thank you David, Leonid.

/integrate

**Pull request closed without merging**